Automattic/sensei
# Add student progress migration

Resolves #5437 

- This PR doesn't introduce a data migration mechanism. Only an interface for migrations and the migration for student progress comments.
- It doesn't work with transactions at this moment. We decided to consider them later.
- Only a very basic test added for the migration. Tests are WIP.

## Proposed Changes
* Add a migration to copy status comments into custom tables.
* Add a tool to run migration manually.

## Testing Instructions

1. Enable the feature: `add_filter( 'sensei_feature_flag_tables_based_progress', '__return_true' );`.
2. Go to `Sensei LMS -> Tools`.
3. Run `Migrate comment-based progress`.
4. You can run it several times. It processes 1000 comments per run.
5. If you see the message "Progress entries created based on comments: 0." it means there is nothing to migrate already.
6. To run it from the beginning you need to clean at least the option: `DELETE FROM `wp_options` where option_name = 'sensei_migrated_progress_last_comment_id';`
7. However, it won't insert duplicates, so you need to delete progress rows as well: `TRUNCATE TABLE `wp_sensei_lms_progress`;`

## Pre-Merge Checklist
